build/build.go

```diff
@@ -910,19 +948,76 @@ func (s *Session) BuildFiles(filenames []string, pkgObj string, cwd string) erro
 	return s.WriteCommandPackage(archive, pkgObj)
 }
 
-// BuildImportPath loads and compiles package with the given import path.
-//
-// Relative paths are interpreted relative to the current working dir.
-func (s *Session) BuildImportPath(path string) (*compiler.Archive, error) {
-	_, archive, err := s.buildImportPathWithSrcDir(path, "")
-	return archive, err
+// BuildProject builds a command project (one with a main method) or
+// builds a test project (one with a synthesized test main package).
+func (s *Session) BuildProject(pkg *PackageData) (*compiler.Archive, error) {
+	// ensure that runtime for gopherjs is imported
+	pkg.Imports = append(pkg.Imports, `runtime`)
+
+	// Build the project to get the parsed packages.
+	var parsed *ParsedPackage
+	var err error
+	if pkg.IsTest {
+		parsed, err = s.buildTestPackage(pkg)
+	} else {
+		parsed, err = s.buildPackages(pkg)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO(grantnelson-wf): At this point we have all the parsed packages we
+	// need to compile the whole project, including testmain, if needed.
+	// We can perform analysis on the whole project at this point to propagate
+	// flatten, blocking, etc. information and check types to get the type info
+	// with all the instances for all generics in the whole project.
+
+	return s.compilePackages(parsed)
+}
+
+func (s *Session) buildTestPackage(pkg *PackageData) (*ParsedPackage, error) {
```

Comment:
Does this method actually perform compilation (directly or indirectly)? I don't think it does, in which case it may be better to name it something like `loadTestPackage`. Same consideration applies to the other sibling methods.

Reply:
Renamed:
- `buildTestPackage` -> `loadTestPackage`
- `buildPackages` -> `loadPackages`
- `buildImportPathWithSrcDir` -> `loadImportPathWithSrcDir`

Reply:
Thank you!